### PR TITLE
Thing proxy fixes

### PR DESCRIPTION
**Pull request by esdrasjnr**
What does this implement/fix? Explain your changes
---------------------------------------------------
This patch fixes some bugs in thing_proxy file, adding a error checking and fixing the Thing structure.

Where has this been tested?
---------------------------

**Operating System/Platform:** Ubuntu 18.04.4

**Go Version:** 1.13.6

### CODE REVIEW

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

File: pkg/thing/delivery/http/thing_proxy.go [hunk 1]
```diff
@@ -231,8 +231,15 @@ func (p proxy) getPaginatedThings(authorization string) ([]*ThingProxyRepr, erro
 			p.logger.Error(err)
 			return nil, err
 		}
+
```
Comment (lcbm) [inline pkg/thing/delivery/http/thing_proxy.go]: In order to maintain code standardization throughout the project, I wouldn't add a  blank line between the 'defer' statement and `keepGoing`. It also seems that other projects such as [mainflux](https://github.com/mainflux/mainflux/blob/master/sdk/go/sdk.go) and [Go](https://github.com/golang/go/search?l=Go&q=defer) itself, seem to not add a line between the `defer` statement and the function it regards to.

File: pkg/thing/delivery/http/thing_proxy.go [hunk 2]
```diff
@@ -267,7 +267,7 @@ func (p proxy) GetThing(authorization, ID string) (*entities.Thing, error) {
 	for i := range things {
 		t := things[i]
 		if t.Metadata.Knot.ID == ID {
-			nt := &entities.Thing{ID: t.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema}
+			nt := &entities.Thing{ID: ID, Token: t.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema}
```
Comment (lcbm) [inline pkg/thing/delivery/http/thing_proxy.go:269]: Would **line 210** also need this modification? :thinking: 
> **210** `things = append(things, &entities.Thing{ID: t.Metadata.Knot.ID, Name: t.Name, Schema: t.Metadata.Knot.Schema})`

**Review by lcbm — COMMENTED** [review]
Awesome catch with the changes in this PR :rocket: 

I left a few comments, let me know when you take a look at them :smiley:

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/unregister_thing_test.go  3
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/delivery/http/thing_proxy.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

File: pkg/thing/delivery/http/thing_proxy.go (hunk 1 above)
Comment (esdrasjnr) [reply pkg/thing/delivery/http/thing_proxy.go]: Thanks for advision @lcbm. Due to your comment I also remove another blank lines before `defer` statement in code and also change the commit message. Done!

File: pkg/thing/delivery/http/thing_proxy.go (hunk 2 above)
Comment (esdrasjnr) [reply pkg/thing/delivery/http/thing_proxy.go:269]: Not in this case @lcbm. Note that the `things` variable will be returned to the user as response of a `device.cmd.list` request, and he doesn't need to know the token (mainflux user's service ID) of every device. Also, the KNoT ID value is right (accessed via Metadata).

Comment (lcbm) [reply pkg/thing/delivery/http/thing_proxy.go:269]: Gotcha! Thanks for the thorough explanation :smile:

File: pkg/thing/delivery/http/thing_proxy.go (hunk 1 above)
Comment (lcbm) [reply pkg/thing/delivery/http/thing_proxy.go]: Awesome! :confetti_ball:

**Review by lcbm — APPROVED** [review]
Awesome! Everything looks good to me :100:

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/interactors/unregister_thing_test.go  3
- pkg/thing/handler/amqp/msg_handler.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/interactors/unregister_thing_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/interactors/unregister_thing_test.go  3
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/delivery/http/thing_proxy.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/interactors/unregister_thing_test.go  3
- pkg/thing/delivery/http/thing_proxy.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Review by vitbaq — APPROVED** [review]
LGTM

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/unregister_thing_test.go  3
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/interactors/unregister_thing.go  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity increasing per file
==============================
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/delivery/http/thing_proxy.go  1
- pkg/thing/interactors/unregister_thing.go  7
- pkg/thing/interactors/unregister_thing_test.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/delivery/http/thing_proxy.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Review by netoax — CHANGES_REQUESTED** [review]
_e68fc77e6b7b66d909cc869e1bf169224a34b7ce_
Start the last part of the commit title with a capital letter to follow our standard.

**thing: proxy: Fix pagination method**

When a **request is sent** to thing's service, the response should be
checked to confirm that everything went well. This patch guarantees that
the returned status code **does** not indicate an error. **It also** removes blank
lines before defer statements.

_aaff40040b243a2019643e1c9f7ecc9cbf24ccd8_
Same comment I did previously about the title and notice the `GetThing` method was renamed.

**Comment by netoax** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/delivery/http/thing_proxy.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5092440&bid=16751168)

**Review by netoax — APPROVED** [review]

**Pull request merged**